### PR TITLE
Enable specifying api key in config

### PR DESCRIPTION
**Pull request by joewagner**
This enables specifying an api key that bypasses the Validator's rate limiting.
You can pass an `apiKey` parameter to the `Validator` and `Database` classes, which is converted to the correct header and send with read requests.

Example use:
```
new Validator({ apiKey, ... });
new Database({ apiKey, ... });
```

### CODE REVIEW

File: packages/local/package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
   "scripts": {
     "postinstall": "cd registry && npm install",
     "test": "mocha",
-    "test:ci": "npm run build && mocha --exit",
```
Comment (joewagner) [inline packages/local/package.json:48]: Not related to this PR, but lerna and nx handle the need to build (or not) before running the test.  Having this written into the test command was causing issues.

File: packages/sdk/package.json [hunk 2]
```diff
@@ -59,7 +59,8 @@
   },
   "scripts": {
     "prepublishOnly": "npm run build",
-    "test": "npm run test:browser && mocha --exit",
+    "test": "npm run test:browser && npm run test:mocha",
+    "test:mocha": "mocha --exit",
```
Comment (joewagner) [inline packages/sdk/package.json:63]: I added this command to make it easier for devs to just run the mocha tests since the browser tests take so long.

File: packages/sdk/src/helpers/config.ts [hunk 3]
```diff
@@ -1,11 +1,13 @@
 import fs from "node:fs";
 import { type WaitableTransactionReceipt } from "../registry/utils.js";
+import { type FetchConfig } from "../validator/client/index.js";
 import { type ChainName, getBaseUrl } from "./chains.js";
 import { type Signer, type ExternalProvider, getSigner } from "./ethers.js";
 
 export interface ReadConfig {
   baseUrl: string;
   aliases?: AliasesNameMap;
+  apiKey?: string;
```
Comment (joewagner) [inline packages/sdk/src/helpers/config.ts:10]: The usage is now:
```
new Validator({ apiKey, ... });
new Database({ apiKey, ... });
```

File: packages/sdk/src/helpers/config.ts [hunk 4]
```diff
@@ -98,3 +100,16 @@ export function jsonFileAliases(filepath: string): AliasesNameMap {
     },
   };
 }
+
+export function prepReadConfig(config: Partial<ReadConfig>): FetchConfig {
+  const conf: FetchConfig = {};
+  if (config.apiKey) {
+    conf.init = {
+      headers: {
+        "Api-Key": config.apiKey,
```
Comment (joewagner) [inline packages/sdk/src/helpers/config.ts:109]: This is the bulk of the logic.  Take the `apiKey` param and set the `Api-Key` request header with the given value.

File: packages/sdk/src/validator/query.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { type Signal } from "../helpers/await.js";
+import { type Signal } from "../helpers/index.js";
```
Comment (joewagner) [inline packages/sdk/src/validator/query.ts:1]: unrelated cleanup

File: packages/sdk/test/validator/config.json [hunk 6]
```diff
@@ -3,7 +3,8 @@
   "HTTP": {
     "Port": "8081",
     "RateLimInterval": "1s",
-    "MaxRequestPerInterval": 10
+    "MaxRequestPerInterval": 10,
+    "APIKey": "foo"
```
Comment (joewagner) [inline packages/sdk/test/validator/config.json:7]: Keeping the tests simple for the time being.  The api key is always going to be "foo"

As an aside in regard to the api key implementation in the validator... Since the key is a fixed config value we should probably ensure the value is only ever used on the server side of the Studio.
Otherwise anyone who logs into the Studio will be able to get the api key and use it as much as they like.

**Comment by carsonfarmer** [comment]
Is this still WIP @joewagner?

**Comment by joewagner** [comment]
> Is this still WIP @joewagner?

It's ready for review, but it's aiming at the cleanup branch, which I'll merge and then point this at main.

File: packages/sdk/test/validator/config.json (hunk 6 above)
Comment (asutula) [reply packages/sdk/test/validator/config.json:7]: Vercel makes it hard to expose secrets to the client. By default all env vars are server side only. You have to opt in to exposing them to the client by prefixing the env var name with `NEXT_PUBLIC_`.

**Review by asutula — APPROVED** [review]
This looks great. Nice work. I'm fine with this as is, but just want to surface an idea... What about making the config option called `extraHeaders` and it's type be `{ [key: string]: string }`. I'm just thinking this is, at this point at least, kind of a bandaid fix for Studio and not a real feature we want to advertise. If we leave the config option as kinda generic like that, we don't have to explain much about it in the docs.
